internal/process/provisioning/create_runtime_resource_step.go

```diff
@@ -229,19 +228,19 @@ func (s *CreateRuntimeResourceStep) providerValues(operation *internal.Operation
 			ProvisioningParameters: operation.ProvisioningParameters,
 		}
 	case broker.FreemiumPlanID:
-		var freemiumProvider internal.CloudProvider
-		if operation.ProvisioningParameters.Parameters.Provider == nil {
-			return provider.Values{}, fmt.Errorf("provider not defined for freemium plan")
-		}
-		freemiumProvider = *operation.ProvisioningParameters.Parameters.Provider
-		switch freemiumProvider {
+		switch operation.ProvisioningParameters.PlatformProvider {
```

Comment:
provider comes from the kind of a BTP account, not a parameter set by a customer